website/blog/2024-04-15-0.74-post.md

```diff
@@ -0,0 +1,407 @@
+
+If the tapped notification resulted in app launch, call `setInitialNotification:`. If the notification was not previously handled by `userNotificationCenter:willPresentNotification:withCompletionHandler:`, call `didReceiveNotification:` as well:
+
+```objectivec
+- (void)  userNotificationCenter:(UNUserNotificationCenter *)center
+  didReceiveNotificationResponse:(UNNotificationResponse *)response
+           withCompletionHandler:(void (^)(void))completionHandler
+{
+  // This condition passes if the notification was tapped to launch the app
+  if ([response.actionIdentifier isEqualToString:UNNotificationDefaultActionIdentifier]) {
+  // Allow the notification to be retrieved on the JS side using getInitialNotification()
+    [RCTPushNotificationManager setInitialNotification:response.notification];
+  }
+ // This will trigger 'notification' and 'localNotification' events on PushNotificationIOS
+  [RCTPushNotificationManager didReceiveNotification:response.notification];
+  completionHandler();
+}
+```
+
+Finally, delete the following methods and adapt the logic into the callbacks above which will be called instead:
+
+1. <code>[application:didReceiveLocalNotification:](https://developer.apple.com/documentation/uikit/uiapplicationdelegate/1622930-application?language=objc)</code> [deprecated]
+2. <code>[application:didReceiveRemoteNotification:](https://developer.apple.com/documentation/uikit/uiapplicationdelegate/1623117-application?language=objc)</code> [deprecated]
+3. <code>[application:didReceiveRemoteNotification:fetchCompletionHandler:](https://developer.apple.com/documentation/uikit/uiapplicationdelegate/1623013-application?language=objc)</code> [not deprecated, but is superseded by the <code>UNUserNotificationCenterDelegate</code> methods]
+
+Delete any usages of <code>[application:didRegisterUserNotificationSettings:](https://developer.apple.com/documentation/uikit/uiapplicationdelegate/1623022-application?language=objc)</code> and <code>RCTPushNotificationManager</code>’s corresponding <code>didRegisterUserNotificationSettings:</code> as well.
+
+**Example:** See the RNTester <code>[AppDelegate.mm](https://github.com/facebook/react-native/blob/main/packages/rn-tester/RNTester/AppDelegate.mm)</code>.
+
+#### JS
+
+1. Remove any references to `alertAction`.
+2. Remove the `handler` argument on any calls to `removeEventListener`.
+3. Replace any usages of `repeatInterval` by firing multiple notifications using `fireDate` or `fireIntervalSeconds` instead.
+4. Note that `soundName` will be null when it is accessed on a `Notification` returned from `getScheduledLocalNotifications()` and `getDeliveredNotifications()`.
+
```

Comment:
@huntie On Android, the apps are now occupying ~3Mb less space on device (it's nearly 10% less). I'd like to add a short paragraph about it.